lib/src/test/java/io/ably/lib/test/realtime/ConnectionManagerTest.java

```diff
@@ -562,37 +565,49 @@ public void onConnectionStateChanged(ConnectionStateChange state) {
 
 			/* Attach to channel */
 			final Channel channel = ably.channels.get(channelName);
+			channel.on(new ChannelStateListener() {
+				@Override
+				public void onChannelStateChanged(ChannelStateChange stateChange) {
+					System.out.println("======================= state is " + stateChange.current);
+				}
+			});
 			channel.once(ChannelState.attached, new ChannelStateListener() {
 				@Override
 				public void onChannelStateChanged(ChannelStateChange stateChange) {
 					System.out.println("Channel attached for the first time");
 					assertEquals("Channel is attached", stateChange.current, ChannelState.attached);
 				}
 			});
+
+			/* attach and wait for the channel to be attached */
 			channel.attach();
-			(new Helpers.ChannelWaiter(channel)).waitFor(ChannelState.attached);
+			ChannelWaiter channelWaiter = new Helpers.ChannelWaiter(channel);
+			channelWaiter.waitFor(ChannelState.attached);
 
-			ably.connection.once(ConnectionEvent.disconnected, new ConnectionStateListener() {
-				@Override
-				public void onConnectionStateChanged(ConnectionStateChange state) {
-					synchronized (ably.connection.connectionManager) {
-						try {
-							/* The client will try to reconnect right away after disconnection.
-							 * We want it to stay into a disconnected state long enough
-							 * so that the connection becomes stale.
-							 */
-							ably.connection.connectionManager.wait(waitInDisconnectedState);
-						} catch (InterruptedException e) {
-							e.printStackTrace();
-						}
-					}
-				}
-			});
+			/* suppress automatic retries by the connection manager */
+			try {
+				Method method = ably.connection.connectionManager.getClass().getDeclaredMethod("disconnectAndSuppressRetries");
+				method.setAccessible(true);
+				method.invoke(ably.connection.connectionManager);
+			} catch (NoSuchMethodException|IllegalAccessException|InvocationTargetException e) {
+				fail("Unexpected exception in suppressing retries");
+			}
 
-			ably.connection.connectionManager.requestState(ConnectionState.disconnected);
 			connectionWaiter.waitFor(ConnectionState.disconnected);
+			assertEquals("Disconnected state was not reached", ConnectionState.disconnected, ably.connection.state);
+			System.out.println("=============================== DISCONNECTED ");
```

Comment:
What are these for lines for?

Reply:
That were meant for debugging but have been deleted in this commit https://github.com/ably/ably-java/pull/390/commits/6ab31f7b3aaffcd4bb46e4ca7133cb92e1c69d0e#diff-4cfebe2518c34c9f3975cde19f3ced08L598